yy_parser.go

```diff
@@ -139,6 +140,9 @@ func (parser *Parser) SetParserConfig(config ParserConfig) {
 func (parser *Parser) Parse(sql, charset, collation string) (stmt []ast.StmtNode, warns []error, err error) {
 	if charset == "" {
 		charset = mysql.DefaultCharset
+	} else {
+		enc, _ := chs.Lookup(charset)
```

Comment:
Do we need to filter out character sets that are not currently supported?

Reply:
We had better filter them in TiDB because it is also used by other users.